src/dataset-builder/CohortEditor.ts

```diff
@@ -396,8 +396,19 @@ export const CriteriaGroupView: React.FC<CriteriaGroupViewProps> = (props) => {
                 Link,
                 {
                   'aria-label': 'delete group',
-                  onClick: () =>
-                    updateCohort(_.set('criteriaGroups', _.without([criteriaGroup], cohort.criteriaGroups))),
+                  onClick: () => {
+                    updateCohort(_.set('criteriaGroups', _.without([criteriaGroup], cohort.criteriaGroups)));
+                    if (cohort.criteriaGroups.indexOf(criteriaGroup) < cohort.criteriaGroups.length - 1) {
+                      // update names of all groups after it
+                      for (
+                        let i = cohort.criteriaGroups.indexOf(criteriaGroup) + 1;
+                        i < cohort.criteriaGroups.length;
+                        i++
+                      ) {
+                        cohort.criteriaGroups[i].name = `Group ${i}`;
```

Comment:
That said, aside from the comment I left above, I think @pshapiro4broad has the right of it that updating the rendering code is probably the right call, rather than setting `criteriaGroup.name`

Reply:
https://github.com/DataBiosphere/terra-ui/pull/4958/files#diff-1c7f8a75f8b65b162da35f364969f231d11a338b08006b900f929d22618ff2d6R394 is the line that renders the name, so this would need to change to use the index, which would probably need to be passed in via props

Reply:
Ah okay, thanks so much for the feedback!! I'll look into making those changes